design/ceph/object/store.md

```diff
@@ -221,6 +221,293 @@ spec:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: glacier-data-pool
+  namespace: rook-ceph
+spec:
+  ...
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: reduced-data-pool
+  namespace: rook-ceph
+spec:
+  ...
+```
+
+Then update `CephObjectStore` CRD  `sharedPools` section with target placements and StorageClasses definitions:
+
+```yaml
+spec:
+  # [OPTIONAL] - sharedPools is mutually exclusive with spec.metadataPool and spec.dataPool
+  # It is the place to tell RGW which pools can be used to store buckets and objects data and metadata.
+  # All pools under this section should be created beforehand.
+  # Rook will append rados namespace to each of the pools, so it is safe to use the same pool across multiple RGW instances.
+  # It is also safe to use the same pool in different target placements.
+  sharedPools:
+    #  [EXISTING,OPTIONAL] - metadata pool. Will be used by default to store bucket index
+    # default poolPlacement.metadataPoolName will override this value, so it is not makes sense to use together with poolPlacements.
+    metadataPoolName: rgw-meta-pool # results to rgw-meta-pool:<CephObjectStore.name>.index
+
+    #  [EXISTING,OPTIONAL] - data pool. Will be used by default to store bucket objects
+    # default poolPlacement.dataPoolName will override this value, so it is not makes sense to use together with poolPlacements.
+    dataPoolName: rgw-data-pool # results to rgw-data-pool:<CephObjectStore.name>.data
+
+    # New settings:
+    # [OPTIONAL] - defines pool placement targets.
+    # Placement targets control which Pools are associated with a particular bucket.
+    # See: https://docs.ceph.com/en/latest/radosgw/placement/#placement-targets
+    poolPlacements:
+      # Default pool placement:
+      - name: "default"                         # [REQUIRED] - placement id. Placement with name "default" will be used as default,
+                                                # and therefore will override sharedPools.metadataPoolName and sharedPools.dataPoolName if presented.
+                                                # This means that defaul poolPlacement is also mutually exclusive with spec.metadataPool and spec.dataPool
+        metadataPoolName: rgw-meta-pool         # [REQUIRED] results to rgw-meta-pool:<CephObjectStore.name>.index
+        dataPoolName: rgw-data-pool             # [REQUIRED] results to rgw-data-pool:<CephObjectStore.name>.data
+        dataNonECPoolName: ""                   # [OPTIONAL] results to rgw-data-pool:<CephObjectStore.name>.data.nonec
+        # [OPTIONAL] - Storage classes for cold placement target.
+        storageClasses:
+          - name: GLACIER                       # [REQUIRED] StorageClass name. Can be arbitrary but it is better to use one of AWS x-amz-storage-class names.
```

Comment:
Do we need a storage class for the default placement? Or shall we stick with `STANDARD`? It's odd to use glacier as an example for the default, since glacier is usually a reduced redundancy cold storage.

Reply:
Moreover some clients / libs expect specific behavior from a storageclass named `GLACIER` so in the Cephiverse we should avoid that name entirely.  The default SC should indeed be `STANDARD`.  Admins can assign a different default for a given user or bucket if they need to.  For an example of a non-default SC, use `STANDARD-IA`.

Reply:
I see why it might be confusing but it is not default StorageClass. It is additional StorageClass available for default placement. All Placements have default class `STANDARD` which always routes to `placement.dataPool` and this cannot be changed. With `placement.storageClasses` it is possible to list additional storage classes which will override `dataPool` of the placement if user selects this class during object creation. Please check [resulted zone.json here](https://github.com/rook/rook/pull/14455/files#diff-cf542a1a819df09579569b1ca4896a8c8530c9437a1ee8f361063b505767aa3dR396-R419).
StorageClasses are defined per placement, so if someone wants to add a new storageClass `GLACIER` it is makes sense to add it to all placements in the list, especially including default placement.

RGW allows to change default storageClass but [it can be done only per user in user settings](https://docs.ceph.com/en/latest/radosgw/placement/#user-placement). So we can add this config to `CephObjectStoreUser` if needed.

Is it makes sense to rename `storageClasses` to `additionalStorageClasses` to avoid such confusion?

Reply:
I'd suggest to keep the `storageClasses` name, and solve this with some changes to the examples and comments:
- Currently the `storageClasses` is included as an example with all of these placements, so it looks more required. How about only include `storageClasses` with one of the placements (perhaps with `hot`)?
- The placements without a storageClass can have a comment that indicates they still get the default `STANDARD` storageClass.